transcom/mymove
# INT-B-21505 prime port update

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1018852&RoomContext=TeamRoom%3A1007470&concept=TeamRoom)

## Summary
This story is to allow the prime user to update the port location for a move with either a POEFSC or PODFSC service item.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
Note: the functionality for ordering POEFSC and PODFSC service items does not yet exist. Therefore in order to test you will need to manually create either a POEFSC or PODFSC service item for a shipment.

1. Create an INTL UB shipment as a customer (CONUS -> AK or AK -> CONUS)
2. Manually create the mto_service_items table for the shipment: `
INSERT INTO mto_service_items (
					id,
                    mto_shipment_id,
                    move_id,
                    re_service_id,
                    status,
                    created_at,
                    updated_at,
                    approved_at
                )
                VALUES (
                uuid_generate_v4(),
                    '8eeba537-bb40-4f58-a534-74ea9ff64590',
                    '813b7a43-2e3f-42f3-8c4e-eca95390d5ca',
                    '388115e8-abe9-441d-96cf-a39f24baa0a3',
                    'APPROVED'::service_item_status,
                    NOW(),
                    NOW(),
                    NOW()
                );
`
3. In the prime v3 api, call getMoveTaskOrder using the move_id for the service item you created in step 2. You will need the etag for the newly created service item for the next step.
4. In the prime v1 api, call updateMTOServiceItem using a payload like this (adjust for you specific service item ID): `
{
  "id": "40b7e71a-6b03-4413-8601-dea694253ee3",
  "modelType": "UpdateMTOServiceItemInternationalPortFSC",
  "portCode": "PDX",
  "reServiceCode": "PODFSC"
}
`
5. Verify that you receive a 200 response like this: `{
  "reServiceCode": "PODFSC",
  "eTag": "MjAyNC0xMi0wMlQxNjo0MjoxNi42NjUwNDla",
  "id": "40b7e71a-6b03-4413-8601-dea694253ee3",
  "modelType": "MTOServiceItemBasic",
  "moveTaskOrderID": "813b7a43-2e3f-42f3-8c4e-eca95390d5ca",
  "mtoShipmentID": "8eeba537-bb40-4f58-a534-74ea9ff64590",
  "reServiceName": "International POD Fuel Surcharge",
  "status": "APPROVED"
}`
6. In the prime v3 api, call getMoveTaskOrder using the same move_id as before.
7. Verify that the updated shipment now has the portOfDebarkation (or portOfEmbarkation) object as part of the response like this: 
`"portOfDebarkation": {
        "city": "PORTLAND",
        "country": "UNITED STATES",
        "county": "MULTNOMAH",
        "id": "b6e94f5b-33c0-43f3-b960-7c7b2a4ee5fc",
        "portCode": "PDX",
        "portName": "PORTLAND INTL",
        "portType": "A",
        "state": "OREGON",
        "zip": "97220"
      },`


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
